test/parallel/test-net-connect-local-error.js

```diff
@@ -3,25 +3,40 @@ const common = require('../common');
 const assert = require('assert');
 const net = require('net');
 
+function getUnassignedPort(cb) {
+  const server = net.createServer();
+  server.listen({port: 0}, common.mustCall(() => {
+    // When the server is closed this port will no longer be assigned
+    const unassignedPort = server.address().port;
+    server.close(common.mustCall(() => {
```

Comment:
TBH I don't see a completely safe way of calling `net.connect()` to a free port. I would probably just move [the original test (before the port changes)](https://github.com/sebastianplesciuc/node/blob/eeae3bd07145a770209e4899a9d40f67109d3d01/test/parallel/test-net-connect-local-error.js) to sequential.
/cc @nodejs/testing @thefourtheye